parser/wasm_parse/mod.rs

```diff
@@ -73,7 +73,9 @@ impl<'a> Parse<'a> for wasmparser::ModuleReader<'a> {
         match (function_section, code_section) {
             (Some(function_section), Some(code_section)) => (function_section, code_section)
                 .parse_items(items, (imported_functions, &names, &sizes))?,
-            _ => panic!("function or code section is missing"),
+            _ => Err(traits::Error::with_msg(
+                "function or code section is missing",
+            ))?,
```

Comment:
Ah ok perfect -- I think `failure::bail!` wouldn't work for us anyways since I forgot that we don't use `failure::Error`